aspnetcore/hosting/ihostingstartup.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Add app features from an external assembly using IHostingStartup in ASP.NET Core
+author: guardrex
+description: Discover how to add features to an ASP.NET Core app from an external assembly using an IHostingStartup implementation.
+ms.author: riande
+manager: wpickett
+ms.date: 12/07/2017
+ms.topic: article
+ms.custom: mvc
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: hosting/ihostingstartup
+---
+# Add app features from an external assembly using IHostingStartup in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+An [IHostingStartup](/dotnet/api/microsoft.aspnetcore.hosting.ihostingstartup) implementation allows adding features to an app at startup from outside of the app's `Startup` class. For example, an external tooling library can use an `IHostingStartup` implementation to provide additional configuration providers or services to an app. `IHostingStartup` *is available in ASP.NET Core 2.0 and later.*
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/hosting/ihostingstartup/sample/) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## Discover loaded hosting startup assemblies
+
+To discover hosting startup assemblies loaded by the app or by libraries, enable logging and check the application logs. Errors that occur when loading assemblies are also logged. Loaded hosting startup assemblies are logged at the Debug level, and errors are always logged.
+
+It's also possible to read the value of the [HostingStartupAssembliesKey](/dotnet/api/microsoft.aspnetcore.hosting.webhostdefaults.hostingstartupassemblieskey). The sample app reads the key into a `string` array and displays the result in the app's Index page:
```

Comment:
>It's also possible to read the value of the
You can read the value of the

Reply:
Or better yet, nuke that entire sentence. The following sample reads the HostingStartupAssembliesKey key into ...

Reply:
I'll make the change, but this is an alternative to the first option of looking at the logs. This loses it's conversational tone by jumping right to 'the sample does Y for this' here.